File: platform/android/java/lib/src/com/dodorare/crossbow/plugin/ExposedToCrossbow.java
package com.dodorare.crossbow.plugin;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Annotation to indicate methods that should be exposed to external call.
 *
 * At runtime, annotated plugin methods are detected and automatically registered.
 */
@Target({ ElementType.METHOD })
@Retention(RetentionPolicy.RUNTIME)
public @interface ExposedToCrossbow {}

File: platform/android/java/lib/src/com/crossbow/library/plugin/ExposedToCrossbow.kt
package com.crossbow.library.plugin

import kotlin.annotation.Retention
import kotlin.annotation.AnnotationRetention

/**
 * Annotation to indicate methods that should be exposed to external call.
 *
 * At runtime, annotated plugin methods are detected and automatically registered.
 */
@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER
)
@Retention(AnnotationRetention.RUNTIME)
annotation class ExposedToCrossbow
